internal/pkg/api/server/buffer/dependencies/dependencies.go

```diff
@@ -0,0 +1,346 @@
+	EtcdDefaultConnectionTimeout = 2 * time.Second
+	EtcdKeepAliveTimeout         = 2 * time.Second
+	EtcdKeepAliveInterval        = 10 * time.Second
+	ProjectLockTTLSeconds        = 60
+)
+
+// ForServer interface provides dependencies for Buffer API server.
+// The container exists during the entire run of the API server.
+type ForServer interface {
+	dependencies.Base
+	dependencies.Public
+	ServerCtx() context.Context
+	ServerWaitGroup() *sync.WaitGroup
+	PrefixLogger() log.PrefixLogger
+	EtcdClient(ctx context.Context) (*etcd.Client, error)
+}
+
+// ForPublicRequest interface provides dependencies for a public request that does not contain the Storage API token.
+// The container exists only during request processing.
+type ForPublicRequest interface {
+	ForServer
+	RequestCtx() context.Context
+	RequestID() string
+}
+
+// ForProjectRequest interface provides dependencies for an authenticated request that contains the Storage API token.
+// The container exists only during request processing.
+type ForProjectRequest interface {
+	ForPublicRequest
+	dependencies.Project
+}
+
+// forServer implements ForServer interface.
+type forServer struct {
+	dependencies.Base
+	dependencies.Public
+	serverCtx  context.Context
+	serverWg   *sync.WaitGroup
+	logger     log.PrefixLogger
+	etcdClient dependencies.Lazy[*etcd.Client]
+}
+
+// forPublicRequest implements ForPublicRequest interface.
+type forPublicRequest struct {
+	ForServer
+	logger     log.PrefixLogger
+	requestCtx context.Context
+	requestID  string
+}
+
+// forProjectRequest implements ForProjectRequest interface.
+type forProjectRequest struct {
+	dependencies.Project
+	ForPublicRequest
+	logger log.PrefixLogger
+}
+
+func NewServerDeps(serverCtx context.Context, envs env.Provider, logger log.PrefixLogger, debug, dumpHttp bool) (v ForServer, err error) {
+	// Create tracer
+	var tracer trace.Tracer = nil
+	if telemetry.IsDataDogEnabled(envs) {
+		tracer = telemetry.NewDataDogTracer()
+		_, span := tracer.Start(serverCtx, "kac.lib.api.server.buffer.dependencies.NewServerDeps")
+		defer telemetryUtils.EndSpan(span, &err)
+	}
+
+	// Create wait group - for graceful shutdown
+	serverWg := &sync.WaitGroup{}
+
+	// Get Storage API host
+	storageApiHost := strhelper.NormalizeHost(envs.MustGet("KBC_STORAGE_API_HOST"))
+	if storageApiHost == "" {
+		return nil, errors.New("KBC_STORAGE_API_HOST environment variable is not set")
+	}
+
+	// Create base HTTP client for all API requests to other APIs
+	httpClient := apiHttpClient(envs, logger, debug, dumpHttp)
+
+	// Create base dependencies
+	baseDeps := dependencies.NewBaseDeps(envs, tracer, logger, httpClient)
+
+	// Create public dependencies - load API index
+	startTime := time.Now()
+	logger.Info("loading Storage API index")
+	publicDeps, err := dependencies.NewPublicDeps(serverCtx, baseDeps, storageApiHost)
+	if err != nil {
+		return nil, err
+	}
+	logger.Infof("loaded Storage API index | %s", time.Since(startTime))
+
```

Comment:
Storage API index nepotrebujeme, minimalne nie s komponentami, zatial vobec nie.
Ale neriesil by som to.

Spravis prosim v `cooldown` cycle na to task?
Pridame tam nejaky prepinac, ci sa ma index nacitat na zacitku, alebo `lazy`, ked bude potrebny.
Resp. ci sa maju nacitat iba services, alebo aj components.

Reply:
https://keboola.atlassian.net/browse/BA-73